canopen/objectdictionary/__init__.py

```diff
@@ -29,29 +29,32 @@ def export_od(od, dest: Union[str, TextIO, None] = None, doc_type: Optional[str]
     :rtype: str or None
     """
 
-    doctypes = {"eds", "dcf"}
-    if isinstance(dest, str):
-        if doc_type is None:
-            for t in doctypes:
-                if dest.endswith(f".{t}"):
-                    doc_type = t
-                    break
-
-        if doc_type is None:
-            doc_type = "eds"
-        dest = open(dest, 'w')
-    assert doc_type in doctypes
-
-    if doc_type == "eds":
-        from canopen.objectdictionary import eds
-        return eds.export_eds(od, dest)
-    elif doc_type == "dcf":
-        from canopen.objectdictionary import eds
-        return eds.export_dcf(od, dest)
-
-    # If dest is opened in this fn, it should be closed
-    if type(dest) is str:
-        dest.close()
+    opened_here = False
+    try:
+        doctypes = {"eds", "dcf"}
+        if isinstance(dest, str):
+            if doc_type is None:
+                for t in doctypes:
+                    if dest.endswith(f".{t}"):
+                        doc_type = t
+                        break
+                else:
+                    doc_type = "eds"
+            dest = open(dest, 'w')
+            opened_here = True
+
+        if doc_type == "eds":
+            from canopen.objectdictionary import eds
+            return eds.export_eds(od, dest)
+        elif doc_type == "dcf":
+            from canopen.objectdictionary import eds
+            return eds.export_dcf(od, dest)
+        else:
+            raise NotImplementedError(f"No support for the {doc_type!r} format")
```

Comment:
Hm, I noticed this a little bit too late... this is a little bit unconventional; `NotImplementedError` is not meant for such use. IMO, `ObjectDictionaryError` would be more appropriate.

Reply:
Right. Agreed. It's not always easy to "just fix" something like the close issue and then uncovering other things that needs fixing too. ;)

Reply:
Yeah, one fix quickly leads to the discovery of other issues :) I'll create an issue about this.

Reply:
I created #475.

Reply:
For the record; before this PR, this check was an `assert` (IOW, `AssertionError`, unless running with `-O` or `-OO`).